units/patch_intent.go

```diff
@@ -0,0 +1,241 @@
+package units
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/evergreen-ci/evergreen/model/user"
+	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/validator"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/logging"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const patchIntentJobName = "patch-intent-processor"
+
+func init() {
+	registry.AddJobType(patchIntentJobName,
+		func() amboy.Job { return makePatchIntentProcessor() })
+}
+
+type patchIntentProcessor struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	logger   grip.Journaler
+	env      evergreen.Environment
+
+	Intent     patch.Intent  `bson:"intent" json:"intent"`
+	PatchID    bson.ObjectId `bson:"patch_id" json:"patch_id" yaml:"patch_id"`
+	user       *user.DBUser
+	projectRef *model.ProjectRef
+}
+
+// NewPatchIntentProcessor creates an amboy job to create a patch from the
+// given patch intent with the given object ID for the patch
+func NewPatchIntentProcessor(patchID bson.ObjectId, intent patch.Intent) amboy.Job {
```

Comment:
again, should we keep patchids being bson.ObjectIds?

Reply:
I don't see a good reason to change now.